### PR TITLE
Bluetooth: host: Configurable time to run slave conn param update

### PR DESCRIPTION
**Pull request by ppryga-nordic**
Enable possibility to configure time a slave will wait until
start of connection parameters update procedure after BT connection
is established.

Signed-off-by: Piotr Pryga <piotr.pryga@nordicsemi.no>

### CODE REVIEW

**Comment by carlescufi** [comment]
@ppryga could you please remove internal Jira references from the commit message? This is upstream and so it doesn't belong here.

File: subsys/bluetooth/host/Kconfig [hunk 1]
```diff
@@ -355,9 +355,20 @@ config BT_MAX_PAIRED
...
+	default 3
+	range 1 65535
+
+config BT_CONN_PARAM_UPDATE_TIMEOUT
+	int "Peripheral connection parameter update timeout in milliseconds"
+	default 5000
+	range 1 65535
+
+	help
+	  The value is a timeout used by slave device to wait until it
```
Comment (carlescufi) [inline subsys/bluetooth/host/Kconfig]: s/slave device/the peripheral

File: subsys/bluetooth/host/Kconfig [hunk 2]
```diff
@@ -355,9 +355,20 @@ config BT_MAX_PAIRED
...
+	  starts connection parameters update procedure to change default
```
Comment (carlescufi) [inline subsys/bluetooth/host/Kconfig]: the connection

File: subsys/bluetooth/host/Kconfig [hunk 3]
```diff
@@ -355,9 +355,20 @@ config BT_MAX_PAIRED
...
+	  connection parameters. Default value is set to 5s, to comply
```
Comment (carlescufi) [inline subsys/bluetooth/host/Kconfig]: The default value

**Review by carlescufi — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/host/Kconfig (hunk 1 above)
Comment (ppryga) [reply subsys/bluetooth/host/Kconfig]: corrected

File: subsys/bluetooth/host/Kconfig (hunk 2 above)
Comment (ppryga) [reply subsys/bluetooth/host/Kconfig]: corrected

File: subsys/bluetooth/host/Kconfig (hunk 3 above)
Comment (ppryga) [reply subsys/bluetooth/host/Kconfig]: corrected

**Comment by ppryga** [comment]
> @ppryga could you please remove internal Jira references from the commit message? This is upstream and so it doesn't belong here.

corrected

File: subsys/bluetooth/host/Kconfig (hunk 2 above)
Comment (jhedberg) [reply subsys/bluetooth/host/Kconfig]: I don't see this change yet?

File: subsys/bluetooth/host/Kconfig (hunk 3 above)
Comment (jhedberg) [reply subsys/bluetooth/host/Kconfig]: Here also it still seems to be the same?

File: subsys/bluetooth/host/Kconfig (hunk 2 above)
Comment (ppryga) [reply subsys/bluetooth/host/Kconfig]: now it should be there, sorry for inconvenience

File: subsys/bluetooth/host/Kconfig (hunk 3 above)
Comment (ppryga) [reply subsys/bluetooth/host/Kconfig]: now it should be there

**Review by jhedberg — APPROVED** [review]

**Review by joerchan — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**